doc/rewriting-services.md

```diff
@@ -1,5 +1,50 @@
 # Tips for rewriting legacy services
 
+## Background
+
+The services are in the process of being rewritten to use our new service
+framework (#1584). Meanwhile, the legacy services extend from an abstract
+adapter called [LegacyService][] which provides a place to put the
+`camp.route()` invocation. The wrapper extends from [BaseService][], so it
+supports badge examples via `category`, `examples`, and `route`. Setting `route`
+also enables `create-service-tester.js` to infer a service's base path, reducing
+boilerplate for [creating the tester][creating a tester].
+
+Legacy services look like:
+
+```js
+camp.route(
```

Comment:
should we show this wrapped in a `module.exports = class ExampleService extends LegacyService {}` as that's what almost every example "in the wild" now looks like